generators/server/generator.mjs

```diff
@@ -24,26 +24,102 @@ export default class extends GeneratorBaseEntities {
+      async packageJson({ application: { buildToolMaven, buildToolGradle } }) {
+        if (buildToolMaven) {
+          this.editFile('package.json', content => content.replaceAll('./mvnw', 'mvnw'));
+        }
+        if (buildToolGradle) {
+          this.editFile('package.json', content => content.replaceAll('./gradlew', 'gradlew'));
+        }
 
         this.packageJson.merge({
           scripts: {
             'native-e2e': 'concurrently -k -s first "npm run native-start" "npm run e2e:headless"',
-            'native-package': 'mvnw package -Pnative,prod -DskipTests',
             'prenative-start': 'npm run docker:db:await --if-present && npm run docker:others:await --if-present',
-            'native-start': './target/native-executable',
-            prepare: 'ln -fs ../../mvnw node_modules/.bin',
           },
         });
+        if (buildToolMaven) {
+          this.packageJson.merge({
+            scripts: {
+              'native-package': 'mvnw package -Pnative,prod -DskipTests',
+              'native-start': './target/native-executable',
+              prepare: 'ln -fs ../../mvnw node_modules/.bin',
+            },
+          });
+        } else if (buildToolGradle) {
+          this.packageJson.merge({
+            scripts: {
+              'native-package': 'gradlew nativeCompile -Pprod -x test -x integrationTest',
+              'native-start': './build/native/nativeCompile/native-executable',
+              prepare: 'ln -fs ../../gradlew node_modules/.bin',
+            },
+          });
+        }
       },
 
       async removeFiles() {
         this.deleteDestination('src/main/resources/logback-spring.xml');
         this.deleteDestination('src/test/resources/logback.xml');
       },
 
-      async pomXml({ application: { buildToolMaven } }) {
+      async customizeGradle({ application: { buildToolGradle, reactive, devDatabaseTypeH2Any } }) {
+        if (!buildToolGradle) return;
+
+        this.addGradlePluginToPluginsBlock('org.springframework.experimental.aot', SPRING_NATIVE_VERSION);
+        this.addGradleMavenRepository('https://repo.spring.io/release');
+        this.addGradlePluginManagementRepository('https://repo.spring.io/release');
+
+        if (devDatabaseTypeH2Any) {
+          if (reactive) {
+            this.editFile('build.gradle', contents => contents.replace('implementation "io.r2dbc:r2dbc-h2"', ''));
+          } else {
+            this.editFile('build.gradle', contents => contents.replace('liquibaseRuntime "com.h2database:h2"', ''));
+          }
+        }
+
+        const buildArgs = ['--no-fallback'];
+        let verbose = false;
+        let memory = '';
+        if (process.env.GITHUB_ACTIONS) {
+          verbose = true;
+          memory = process.platform === 'darwin' ? '-J-Xmx13g' : '-J-Xmx7g';
+          buildArgs.push('--verbose', process.platform === 'darwin' ? '-J-Xmx13g' : '-J-Xmx7g');
+        }
```

Comment:
The missing bit to get maven feature parity is external options.
So we can drop this workaround.
https://github.com/jhipster/generator-jhipster-native/blob/fa4d064fb9060627c6d28c92057493910b32e3b7/.github/workflows/jdl.yml#L59

I wonder if it's possible.

Reply:
Sadly they don't tell which can be set via cli :(

```
For options that can be set using command-line, if both DSL and command-line options are present, command-line options take precedence. 
```

Reply:
Ah good to know. Let me check. I think we could do something like -PnativeBuildArgs=-J-Xmx13g....

Reply:
I hope my changes to jdl are correct. I have added following properties: nativeBuildArgs (for nativeCompile) and nativeImageProperties (maybe we use ...Args here too) which can be used via -PnativeBuildArgs=-J-Xmx13g for example